From 90392042ed7d1e6b85d5a29396ff977dbc412e8a Mon Sep 17 00:00:00 2001
From: Kent Russell <kent.russell@amd.com>
Date: Fri, 24 Mar 2017 07:52:15 -0400
Subject: [PATCH 5558/5855] drm/amdkfd: Initialize packet manager for non-HWS

We use the packet manager to create packets, even in SW-scheduled
situations. Initialize it when we aren't using HWS, and destroy it when
we're done.

BUG: KFD-314

Change-Id: Iac778cedc2ebf3f5285d6012b3682ddba0d3eaa5
Signed-off-by: Kent Russell <kent.russell@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index 31b59bc..9accd80 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -789,11 +789,12 @@ static void uninitialize_nocpsch(struct device_queue_manager *dqm)
 static int start_nocpsch(struct device_queue_manager *dqm)
 {
 	init_interrupts(dqm);
-	return 0;
+	return pm_init(&dqm->packets, dqm, dqm->dev->mec_fw_version);
 }
 
 static int stop_nocpsch(struct device_queue_manager *dqm)
 {
+	pm_uninit(&dqm->packets);
 	return 0;
 }
 
-- 
2.7.4

